### PR TITLE
Run all external node pings in parallel

### PR DESCRIPTION
**Pull request by kyhavlov**
This PR changes the node pings to happen fully parallel to one another, to fix an issue where they were delayed with large numbers of nodes by things like ping timeouts and slow responses from the local Consul agent.

### CODE REVIEW

**Comment by banks** [comment]
Took a look through the code and it seems good to me. I need to catch up on the issue that this was designed to fix and the testing as I saw some discussion around that.

File: coordinate_test.go [hunk 1]
```diff
@@ -215,3 +215,66 @@ func TestCoordinate_reapFailedNode(t *testing.T) {
...
+		t.Fatal(err)
+	}
+	defer s.Stop()
+
+	client, err := api.NewClient(&api.Config{Address: s.HTTPAddr})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Register 2 external nodes with external-probe = true.
```
Comment (banks) [inline coordinate_test.go]: 2? Is that a copy/paste error?

**Comment by kyhavlov** [comment]
@banks This loop for the node pings had them all running in serial before, which was an issue when trying to ping 100+ nodes in each update interval (10s by default). The actual time taken to run the ping and update the right data in the KV store was adding up to more time than the interval and causing things to be slower than expected depending on how many nodes you had.

File: coordinate.go [hunk 2]
```diff
@@ -65,35 +70,50 @@ func (a *Agent) updateCoords(nodeCh <-chan []*api.Node) {
...
-			}
-			if err := a.updateNodeCoordinate(node, rtt); err != nil {
-				a.logger.Printf("[WARN] could not update coordinate for node %q: %v", node.Node, err)
-			}
-		} else {
-			a.logger.Printf("[WARN] could not ping node %q: %v", node.Node, err)
-			if err := a.updateFailedNode(node, kvClient, key, kvPair); err != nil {
-				a.logger.Printf("[WARN] error updating node: %v", err)
-			}
+	// Run an ICMP ping to the node.
```
Comment (pearkes) [inline coordinate.go]: nit: could be ICMP or UDP here.

**Pull request merged**